### PR TITLE
Introduce caching mechanism for very large queries.

### PR DESCRIPTION
**Pull request by huebnerr**

### CODE REVIEW

**Comment by huebnerr** [comment]
Addressing issue https://github.com/vertica/vertica-sql-go/issues/43

File: rows.go [hunk 1]
```diff
@@ -70,18 +78,75 @@ func (r *rows) Columns() []string {
...
+
+	for true {
+		sizeBuf := make([]byte, 4)
+
+		if _, err := r.resultCache.Read(sizeBuf); err != nil {
+			if err == io.EOF {
+				if indexCount == 0 {
+					return false
+				}
+				fmt.Printf("reach EOF reading.. setting result data to %d\n", indexCount)
```
Comment (fbernier) [inline rows.go]: Do we want to keep this message?

Comment (huebnerr) [reply rows.go]: For now, I'm okay with keeping that.

**Comment by huebnerr** [comment]
@sitingren, please do a quick review and merge.

**Comment by sitingren** [comment]
Looks good!

**Pull request merged**